cloud/services/disks/service.go

```diff
@@ -24,6 +24,7 @@ import (
 // DiskScope defines the scope interface for a disk service.
 type DiskScope interface {
 	logr.Logger
+	azure.Authorizer
```

Comment:
Looks like `azure.Authorizer` is needed in most of the places where `azure.ClusterDescriber` are needed. Should the `ClusterDescriber` be composed of `Authorizer`?